wordpress-mobile/WordPress-iOS
# Update Reader detail toolbar

Refs #21644
Designs zQnohyMpLzBzQ5jzMMKni3-fi-868_19180

This updates the Reader detail toolbar component, as per the new design. 

• | Light | Dark
-|-|-
Default state | ![Simulator Screenshot - iPhone 15 - 2023-10-14 at 00 23 34](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/28cc5e3b-a576-4df3-a6ab-0c766890499f) | ![Simulator Screenshot - iPhone 15 - 2023-10-14 at 00 21 26](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/fc3304cd-64b2-447b-ba6c-6c463b622162)
Various states | ![Simulator Screenshot - iPhone 15 - 2023-10-14 at 00 23 47](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/3f3c6346-651e-4c1f-8835-5beef070711b) | ![Simulator Screenshot - iPhone 15 - 2023-10-14 at 00 24 05](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/0653f40f-e9ec-40e0-a700-178bad25d8c4)

To apply the new design, I tried to modify the existing toolbar implementation since it looks close enough to the new design. To achieve the vertical arrangement between the button's image and title, I used `UIButton.Configuration`'s `imagePlacement` (which is pretty convenient!).

This PR also fixes some legacy issues:

- Like button animation being triggered when saving a post.
- The disabled comment button appeared way darker than the rest of the disabled icons.
- Disabled comment icon sometimes appears enabled.
- Disabled Like button sometimes appears enabled.

Some notes: 

- I'm still using the previous icons. Using half-new and half-old icons causes misalignments in the toolbar, and they need to be manually aligned with insets, padding, etc.. I'll use the previous icons for now to reduce the complexity. 
- The toolbar height is hardcoded. Perhaps this was intended, so to have fewer "surprises", I'm simply updating the hardcoded value (from previously 50px to now 58px as per design.)

## To test

- Launch Jetpack.
- Ensure that the `readerImprovements` flag is enabled.
- Navigate to the Reader and select any tab.
- Tap one of the posts.
- 🔎  Verify that the new toolbar is displayed.
- Tap on the Like icon.
- 🔎  Verify that the animation works correctly, and that the post is now liked.
- Go back to the stream, and select one post that you've liked before.
- 🔎 Verify that the Like button is in a selected state. 
- Go back to the stream, and select one post that you've saved.
- 🔎 Verify that the Save button is in a selected state.
- 🔎 Verify that tapping Reblog opens a Site selector sheet.
- 🔎 Verify that tapping Comment opens the comment thread.

Additional tests:
- Test in iPad
- Test VoiceOver

## Regression Notes
1. Potential unintended areas of impact
Toolbar buttons when the feature flag is disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
